circuit-types/src/traits.rs

```diff
@@ -394,15 +444,40 @@ impl CircuitVarType for Variable {
         vec![*self]
     }
 
-    fn from_vars<I: Iterator<Item = Variable>>(i: &mut I) -> Self {
+    fn from_vars<I: Iterator<Item = Variable>, C: ConstraintSystem<ScalarField>>(
+        i: &mut I,
+        _cs: &mut C,
+    ) -> Self {
         i.next().unwrap()
     }
 }
 
+impl CircuitVarType for BoolVar {
+    type BaseType = bool;
+
+    fn to_vars(&self) -> Vec<Variable> {
+        vec![(*self).into()]
+    }
+
+    fn from_vars<I: Iterator<Item = Variable>, C: ConstraintSystem<ScalarField>>(
+        i: &mut I,
+        cs: &mut C,
+    ) -> Self {
+        let var = i.next().unwrap();
+        cs.enforce_bool(var).unwrap();
+
+        BoolVar::new_unchecked(var)
```

Comment:
the warning that `new_unchecked` is private is fixed by an analogous PR in `mpc-jellyfish`